fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java

```diff
@@ -110,15 +111,73 @@ public Table getTable(String dbName, String tblName) {
         try (Connection connection = getConnection()) {
             ResultSet columnSet = schemaResolver.getColumns(connection, dbName, tblName);
             List<Column> fullSchema = schemaResolver.convertToSRTable(columnSet);
+            List<Column> partitionColumns = listPartitionColumns(dbName, tblName, fullSchema);
             if (fullSchema.isEmpty()) {
                 return null;
             }
-            return schemaResolver.getTable(ConnectorTableId.CONNECTOR_ID_GENERATOR.getNextId().asInt(),
-                    tblName, fullSchema, dbName, catalogName, properties);
+            JDBCTableName tableKey = JDBCTableName.of(catalogName, dbName, tblName);
+            if (JDBCTableIdCache.getTableIdCache().containsKey(tableKey)) {
+                return schemaResolver.getTable(JDBCTableIdCache.getTableIdCache().get(tableKey),
+                        tblName, fullSchema, partitionColumns, dbName, catalogName, properties);
+            } else {
+                Integer tableId = ConnectorTableId.CONNECTOR_ID_GENERATOR.getNextId().asInt();
+                JDBCTableIdCache.getTableIdCache().put(tableKey, tableId);
+                return schemaResolver.getTable(tableId, tblName, fullSchema, partitionColumns, dbName, catalogName, properties);
+            }
         } catch (SQLException | DdlException e) {
             LOG.warn(e.getMessage());
             return null;
         }
     }
 
+    @Override
+    public List<String> listPartitionNames(String databaseName, String tableName) {
+        try (Connection connection = getConnection()) {
+            return schemaResolver.listPartitionNames(connection, databaseName, tableName);
+        } catch (SQLException e) {
+            throw new StarRocksConnectorException(e.getMessage());
+        }
+    }
+
+    public List<Column> listPartitionColumns(String databaseName, String tableName, List<Column> fullSchema) {
+        try (Connection connection = getConnection()) {
+            List<String> partitionColumnNames = schemaResolver.listPartitionColumns(connection, databaseName, tableName);
+            ImmutableList.Builder<Column> list = ImmutableList.builder();
+            if (partitionColumnNames.size() > 0) {
+                for (String colName : partitionColumnNames) {
+                    for (Column col : fullSchema) {
+                        if (colName.equals(col.getName())) {
```

Comment:
1. The column name comparison in StarRocks should be case-insensitive. 
2. Build a map for columns and lookup the map could be more efficient and clear

Reply:
Okay, I'll try to make some modifications

Reply:
done